flink-kubernetes/src/test/java/org/apache/flink/kubernetes/highavailability/KubernetesStateHandleStoreTest.java

```diff
@@ -1143,4 +1190,31 @@ private TestingLongStateHandleHelper.LongStateHandle addDeletingEntry(
         configMap.getData().put(key, deleting);
         return state;
     }
+
+    private CompletableFuture<Boolean> retryWithFirstFailedK8sOperation(
+            Function<KubernetesConfigMap, Optional<KubernetesConfigMap>> function,
+            KubernetesConfigMap leaderConfigMap) {
+        final AtomicInteger retryNum = new AtomicInteger(0);
```

Comment:
```suggestion
        final AtomicInteger callbackInvocationCount = new AtomicInteger(0);
```
nit: IMHO, the naming is a bit misleading here: We're not counting the retries (which would be `1` in the end) but the number of times the passed `function` is called.